include/rmm/detail/dynamic_load_runtime.hpp

```diff
@@ -110,17 +104,28 @@ struct dynamic_load_runtime {
  * This allows RMM users to compile/link against CUDA 11.2+ and run with
  * < CUDA 11.2 runtime as these functions are found at call time.
  */
+
+extern "C" {
+cudaError_t cudaMemPoolCreate(cudaMemPool_t*, const cudaMemPoolProps*) __attribute((weak));
+cudaError_t cudaMemPoolSetAttribute(cudaMemPool_t, cudaMemPoolAttr, void*) __attribute((weak));
+cudaError_t cudaMemPoolDestroy(cudaMemPool_t) __attribute((weak));
+cudaError_t cudaMallocFromPoolAsync(void**, size_t, cudaMemPool_t, cudaStream_t)
+  __attribute((weak));
+cudaError_t cudaFreeAsync(void*, cudaStream_t) __attribute((weak));
+cudaError_t cudaDeviceGetDefaultMemPool_sig(cudaMemPool_t*, int) __attribute((weak));
```

Comment:
```suggestion
cudaError_t cudaDeviceGetDefaultMemPool(cudaMemPool_t*, int) __attribute((weak));
```

If it turns out we need to keep these, I think we need this.

Reply:
*question*: How do we ensure that these forward declarations are the same as the symbols that the runtime actually offers? If this were linking to driver code, I would advocate using the versioned interfaces, but I don't think the runtime gives the same guarantees.

I think we are "just" lucky so far that nothing changed between cuda 11 and cuda 12 in these interfaces.

Reply:
Correct. Updated

Reply:
> I think we are "just" lucky so far that nothing changed between cuda 11 and cuda 12 in these interfaces.

Same issue existed in the original version. The cuda runtime headers set these as macros that evaluate to the proper versioned value ( e.g. cudaMalloc_v2 or cudaFreeAsync_ptsz )